homeassistant/components/deconz/alarm_control_panel.py

```diff
@@ -150,20 +150,24 @@ def async_update_callback(self) -> None:
     @property
     def state(self) -> str | None:
         """Return the state of the control panel."""
-        return DECONZ_TO_ALARM_STATE.get(self._device.panel)
+        return DECONZ_TO_ALARM_STATE.get(str(self._device.panel))
```

Comment:
Given `self._device.panel` could be `None`, perhaps this needs a check for `None` first.

Reply:
How well does mypy handle the `Literal` to `str` implicit conversion without the cast?

Reply:
Will improve I've the check of panel. Thanks!

Mypy seems to handle it properly. At the very least it doesn't fail :). As a separate step down the line I will replace all literals with enums but I want to complete typing first.

Reply:
Didn't need to use cast once I stopped using .get

Reply:
That's good to know, thanks. All looks good to me.

Reply:
Thanks for your help @chishm 🎉